client/cli/src/lib.rs

```diff
@@ -112,3 +118,134 @@ impl sc_cli::CliConfiguration for PurgeChainCmd {
 		Some(&self.base.database_params)
 	}
 }
+
+/// The `run` command used to run a node.
+#[derive(Debug, StructOpt)]
+pub struct RunCmd {
+	/// The cumulus RunCmd inherents from sc_cli's
+	#[structopt(flatten)]
+	pub base: sc_cli::RunCmd,
+
+	/// Id of the parachain this collator collates for.
+	#[structopt(long)]
+	pub parachain_id: Option<u32>,
+
+	/// Run node as collator.
+	///
+	/// Note that this is the same as running with `--validator`.
+	#[structopt(long, conflicts_with = "validator")]
+	pub collator: bool,
+}
+
+/// A non-redundant version of the `RunCmd` that sets the `validator` field when the
+/// original `RunCmd` had the `colaltor` field.
+/// This is how we make `--collator` imply `--validator`.
+pub struct NormalizedRunCmd {
+	/// The cumulus RunCmd inherents from sc_cli's
+	pub base: sc_cli::RunCmd,
+	/// Id of the parachain this collator collates for.
+	pub parachain_id: Option<u32>,
+}
+
+impl RunCmd {
+	/// Create a NormalizedRunCmd which does not have a redundant `collator` field.
```

Comment:
```suggestion
	/// Create a [`NormalizedRunCmd`] which merges the `collator` cli argument into `validator` to have only one.
```

Reply:
One last call @JoshOrndorff. Does this sounds reasonable in english? :D

Reply:
Sounds great.